### ISSUE TITLE
Use blib/script/ when run under Test::Harness

### ISSUE THREAD

**Issue by grr** [uperl/Test-Script#40]
Both ExtUtils::MakeMaker and Module::Build will stage scripts that are to be installed into `blib/script`. The former can be specified with `EXE_FILES` and the latter by `script_files`. When `make test` or `Build test` or `prove -blib` is run, the staged script should instead be tested. Testing for `$ENV{HARNESS_ACTIVE}` can be used to tell which script location to choose.

**Comment by plicease** [comment]
I agree that testing `blib/script` is probably the right thing to do when run, and this would be a good thing do implement as an opt-in, but doing so by default will probably break existing dists that are expecting the current behavior.
